File: Competitive Programming/DAY-24/roman.py
num_map=[(1000,'M'),(900,'CM'),(500,'D'),(400,'CD'),(100,'C'),
(90,'XC'),(50,'L'),(40,'XL'),(10,'X'),(9,'IX'),(5,'V'),(4,'IV'),(1,'I')]

def nr(num):
    roman=""
    while num>0:
        for i,r in num_map:
            while num>=i:
                roman=roman+r
                num=num-i
    print("The roman numeral is ",roman)
    
j=int(input("Enter Julius Book Count"))
n=int(input("Enter Nancy Book Count"))
value=j+n
nr(value)

File: Competitive Programming/DAY-24/roman.cpp
#include <iostream>

using namespace std;

void iRoman(int n) 
{
    string str_romans[] = {"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"};
    int values[] = {1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1};

    string result = "";

    for (int i = 0; i < 13; ++i)
    {
        while(n - values[i] >= 0)
         {
            result += str_romans[i];
            n -= values[i];
         }
    }

    cout<<"The Roman Numeral is "<<result<<endl;
}

int main()
{
    int j,n,value;
    cout<<"Enter Julius Book count";
    cin>>j;
    cout<<"Enter Nancy book count";
    cin>>n;
    value=j+n;
    iRoman(value);
}
